src/leader_election.rs

```diff
@@ -139,6 +141,27 @@ pub mod ballot_leader_election {
             }
         }
 
+        /// Construct a new BallotLeaderComponent
+        /// # Arguments
+        /// * `cfg` - Hocon configuration used for ble replica.
+        /// * `peers` - Vector that holds all the other replicas.
+        /// * `initial_leader` -  Initial leader which will be elected.
+        pub fn with_hocon(
+            &self,
+            cfg: &Hocon,
+            peers: Vec<u64>,
+            initial_leader: Option<Leader<Ballot>>,
+        ) -> BallotLeaderElection {
+            BallotLeaderElection::with(
+                peers,
+                cfg[PID].as_i64().unwrap_or_default() as u64,
+                cfg[HB_DELAY].as_i64().unwrap_or_default() as u64,
+                cfg[INCREMENT_DELAY].as_i64().unwrap_or_default() as u64,
```

Comment:
Can you put these into a private function? And instead of `unwrap_or_default()` use `expect()` with a message that lets the user know what value failed to be parsed. For example: "Failed to load PID"